keymanager/awskms.go

```diff
@@ -46,29 +49,35 @@ func (k *Kms) GenerateEnvelopeKey(keyID string, secretID string) (EnvelopeKey, e
 }
 
 // Decrypt decrypts the encrypted key.
-func (k *Kms) Decrypt(keyID string, keyCiphertext []byte, secretID string) ([]byte, error) {
-	client, err := newKmsClient(keyID)
+func (k *Kms) Decrypt(ctx context.Context, keyID string, keyCiphertext []byte, secretID string) ([]byte, error) {
+	client, err := newKmsClient(ctx, keyID)
 	if err != nil {
 		return nil, err
 	}
-	do, err := client.Decrypt(&kms.DecryptInput{
-		EncryptionContext: aws.StringMap(map[string]string{
+	do, err := client.Decrypt(ctx, &kms.DecryptInput{
+		EncryptionContext: map[string]string{
 			"SecretName": secretID,
-		}),
+		},
 		CiphertextBlob: keyCiphertext,
 	})
-	return do.Plaintext, err
+	if err != nil {
+		return []byte{}, err
+	}
```

Comment:
this needed to be added because in case of error, `nil` is returned in the `do` variable, causing a panic when calling `do.Plaintext`